src/core/src/Eryph.Core/Genetics/CatletBreeding.cs

```diff
@@ -0,0 +1,301 @@
+﻿using System;
+using System.Linq;
+using Eryph.ConfigModel;
+using Eryph.ConfigModel.Catlets;
+using Eryph.ConfigModel.Variables;
+using LanguageExt;
+using LanguageExt.Common;
+
+using static LanguageExt.Prelude;
+
+namespace Eryph.Core.Genetics;
+
+public static class CatletBreeding
+{
+    /// <summary>
+    /// Breeds the <paramref name="parentConfig"/> with the <paramref name="childConfig"/>.
+    /// </summary>
+    public static Either<Error, CatletConfig> Breed(
+        CatletConfig parentConfig,
+        CatletConfig childConfig) =>
+        from cpu in BreedCpu(parentConfig.Cpu, childConfig.Cpu)
+        from memory in BreedMemory(parentConfig.Memory, childConfig.Memory)
+        from drives in BreedDrives(Seq(parentConfig.Drives), Seq(childConfig.Drives))
+        from networkAdapters in BreedNetworkAdapters(Seq(parentConfig.NetworkAdapters), Seq(childConfig.NetworkAdapters))
+        from networks in BreedNetworks(Seq(parentConfig.Networks), Seq(childConfig.Networks))
+        from variables in BreedVariables(Seq(parentConfig.Variables), Seq(childConfig.Variables))
```

Comment:
Not sure if missing variables actually stop the breeding process - I called the API first with an older compute client that knows nothing about variables - and it did not fail. 

Catlet used:

``` yaml
parent: dbosoft/ubuntu-22.04/starter

memory:
  startup: 1024

variables:
  - name: password
    required: true
    secret: true

fodder: 
 - source: gene:dbosoft/starter-food:linux-starter
   variables: 
   - name: password
     value: "{{ password }}"
```

Reply:
Missing variables will stop the creation of the catlet, but this only happens in the `UpdateCatletSaga` when the fodder genes are fed to the catlet and the variables are substituted.

When you used the old compute client, it did not include the variables both on catlet level and on fodder level. The config, which the server received, looked like this:
```
parent: dbosoft/ubuntu-22.04/starter

memory:
  startup: 1024

fodder: 
 - source: gene:dbosoft/starter-food:linux-starter
```
This is a valid config as the `linux-starter` food contains (default) values for all variables.

Reply:
true, forget about the client side parsing.